rest-service/manager_rest/resource_manager.py

```diff
@@ -488,31 +631,48 @@ def _check_for_any_active_executions(self):
             for e in self.list_executions(is_include_system_workflows=True,
                                           filters=filters).items
         ]
-
-        if executions:
+        should_queue = False
+        # Execution can't currently run because other executions are running,
+        # since `queue` flag is on - we will queue the execution and it will
+        # run when possible
+        if executions and queue:
+            should_queue = True
+
+        elif executions:
             raise manager_exceptions.ExistingRunningExecutionError(
                 'You cannot start a system-wide execution if there are '
                 'other executions running. '
                 'Currently running executions: {0}'
                 .format(executions))
+        return should_queue
 
-    def _check_for_active_system_wide_execution(self):
+    def _check_for_active_system_wide_execution(self, queue):
         filters = {
             'status': ExecutionState.ACTIVE_STATES
         }
+        should_queue = False
         for e in self.list_executions(is_include_system_workflows=True,
```

Comment:
btw this should likely have `get_all_results=True`, otherwise we'll miss executions when there's more than 1000 of them?